draft-ietf-mls-protocol.md

```diff
@@ -3097,7 +3097,49 @@ MLSPlaintext.confirmation_tag =
     MAC(confirmation_key, GroupContext.confirmed_transcript_hash)
 ~~~~~
 
-### External Commits
+### Adding Members to the Group
+
+New members can join the group in two ways. Either by being added by a group
+member, or by adding themselves through an external Commit. In both cases, the
+new members need information to bootstrap their local group state.
+
+~~~
+struct {
+    ProtocolVersion version = mls10;
+    CipherSuite cipher_suite;
+    opaque group_id<0..255>;
+    uint64 epoch;
+    opaque tree_hash<0..255>;
+    opaque confirmed_transcript_hash<0..255>;
+    Extension group_context_extensions<0..2^32-1>;
+    Extension other_extensions<0..2^32-1>;
+    MAC confirmation_tag;
+    KeyPackageRef signer;
+    opaque signature<0..2^16-1>;
+} GroupInfo;
+~~~
+
+New members MUST verify the `signature` using the public key taken from the
+credential in the leaf node of the member with KeyPackageRef `signer`. The
+signature covers the following structure, comprising all the fields in the
+GroupInfo above `signature`:
+
+~~~
+struct {
+    CipherSuite cipher_suite;
+    opaque group_id<0..255>;
+    uint64 epoch;
+    opaque tree_hash<0..255>;
+    opaque confirmed_transcript_hash<0..255>;
+    Extension group_context_extensions<0..2^32-1>;
+    Extension other_extensions<0..2^32-1>;
+    MAC confirmation_tag;
+    optional<HPKEPublicKey> external_pub;
```

Comment:
Since it's now an extension, do we still need it here?

Reply:
Ah, sorry. Forgot to remove that one. Thanks!